### PR TITLE
Test cleanups

### PR DESCRIPTION
**Pull request by CarysT**
Uhm, the commits are all messed up i think

### CODE REVIEW

File: test/GameSecretTest.cs [hunk 1]
```diff
@@ -101,43 +101,16 @@ public void TestToBytes()
...
-				WasGivenFreeRing = true
-			};
-
-			Assert.AreNotEqual(DesiredSecret, s2);
-			Assert.AreNotEqual(DesiredSecret, null);
-			Assert.AreNotEqual(DesiredSecret, "");
+			Assert.AreNotEqual(DesiredSecret, new GameSecret());
+			Assert.False(new GameSecret().Equals(new TestSecret()));
+			Assert.False(new GameSecret().Equals(new MemorySecret()));
+			Assert.False(new GameSecret().Equals(null));
```
Comment (kabili207) [inline test/GameSecretTest.cs]: You have a mix of code styles here. For consistency with existing code, use the `Assert.AreNotEqual` method instead, unless you want to rewrite everything to use the newer `Assert.That` style documented here: https://github.com/nunit/docs/wiki/Assertions

File: test/MemorySecretTest.cs [hunk 2]
```diff
@@ -108,18 +123,10 @@ public void TestEquals()
...
-				IsReturnSecret = true
-			};
-
-			Assert.AreNotEqual(desiredSecret, s2);
-			Assert.AreNotEqual(desiredSecret, null);
-			Assert.AreNotEqual(desiredSecret, "");
+			Assert.AreNotEqual(desiredSecret, new MemorySecret());
+			Assert.False(new MemorySecret().Equals(new TestSecret()));
+			Assert.False(new MemorySecret().Equals(new GameSecret()));
+			Assert.False(new MemorySecret().Equals(null));
```
Comment (kabili207) [inline test/MemorySecretTest.cs]: You have a mix of code styles here. For consistency with existing code, use the `Assert.AreNotEqual` method instead, unless you want to rewrite everything to use the newer `Assert.That` style documented here: https://github.com/nunit/docs/wiki/Assertions

File: test/SecretParserTest.cs [hunk 3]
```diff
@@ -23,13 +22,22 @@ public class SecretParserTest
 			37, 32, 58, 47, 25
 		};
 
+		private static readonly byte[] highByte = new byte[] { 64 };
+		private static readonly byte[] lowByte = new byte[] { 0 };
```
Comment (kabili207) [inline test/SecretParserTest.cs]: These variables are not used. Remove them.

File: test/SecretParserTest.cs [hunk 4]
```diff
@@ -23,13 +22,22 @@ public class SecretParserTest
...
+
 		[Test]
 		public void CreateString()
 		{
 			string testString = SecretParser.CreateString(desiredSecretBytes, GameRegion.US);
 			Assert.AreEqual(desiredSecretString, testString);
 		}
 
+		[Test]
+		public void TestDivideThrows()
```
Comment (kabili207) [inline test/SecretParserTest.cs]: This is an odd name for what it really checks. Consider renaming.

File: test/GameSecretTest.cs [hunk 5]
```diff
@@ -101,43 +101,16 @@ public void TestToBytes()
...
-				Child = "Pip",
-				Animal = Animal.Dimitri,
-				Behavior = 4,
-				IsLinkedGame = true,
-				IsHeroQuest = false,
-				WasGivenFreeRing = true
-			};
-
-			Assert.AreEqual(DesiredSecret, s2);
+			Assert.True(new GameSecret().Equals(new GameSecret()));
```
Comment (kabili207) [inline test/GameSecretTest.cs]: For consistency with existing code, use the `Assert.AreEqual` method instead, unless you want to rewrite everything to use the newer `Assert.That` style documented here: https://github.com/nunit/docs/wiki/Assertions

**Comment by CarysT** [comment]
Implemented suggested changes, please let me know if anything needs changing.

File: test/MemorySecretTest.cs [hunk 6]
```diff
@@ -59,6 +59,14 @@ public void LoadSecretFromString()
 			Assert.AreEqual(desiredSecret, secret);
 		}
 
+		[Test]
+		public void LoadSecretFromString_JP()
+		{
+			var secret = new MemorySecret();
+			secret.Load(desiredSecretString_JP, GameRegion.JP);
+			Assert.That(desiredSecret_JP.Equals(secret), Is.True);
```
Comment (kabili207) [inline test/MemorySecretTest.cs]: Use `Is.EqualTo`

File: test/MemorySecretTest.cs [hunk 7]
```diff
@@ -73,6 +81,13 @@ public void TestToString()
 			Assert.AreEqual(desiredSecretString, secret);
 		}
 
+		[Test]
+		public void TestToString_JP()
+		{
+			string secret = desiredSecret_JP.ToString();
+			Assert.That(desiredSecretString_JP.Equals(secret), Is.True);
```
Comment (kabili207) [inline test/MemorySecretTest.cs]: Use `Is.EqualTo`

File: test/MemorySecretTest.cs [hunk 8]
```diff
@@ -108,18 +123,10 @@ public void TestEquals()
...
-				TargetGame = Game.Ages,
-				GameID = 14129,
-				Memory = Memory.GraveyardFairy,
-				IsReturnSecret = true
-			};
-
-			Assert.AreNotEqual(desiredSecret, s2);
-			Assert.AreNotEqual(desiredSecret, null);
-			Assert.AreNotEqual(desiredSecret, "");
+			Assert.That(desiredSecret.Equals(new MemorySecret()), Is.False);
```
Comment (kabili207) [inline test/MemorySecretTest.cs]: Use `Is.Not.EqualTo`

File: test/GameSecretTest.cs [hunk 9]
```diff
@@ -101,56 +101,26 @@ public void TestToBytes()
...
-				Child = "Pip",
-				Animal = Animal.Dimitri,
-				Behavior = 4,
-				IsLinkedGame = true,
-				IsHeroQuest = false,
-				WasGivenFreeRing = true
-			};
-
-			Assert.AreEqual(DesiredSecret, s2);
+			Assert.That(new GameSecret().Equals(new GameSecret()), Is.True);
```
Comment (kabili207) [inline test/GameSecretTest.cs]: Change these to use `Is.EqualTo` instead.
For example:
```csharp
Assert.That(new GameSecret(), Is.EqualTo(new GameSecret());
```

File: test/GameSecretTest.cs [hunk 10]
```diff
@@ -101,56 +101,26 @@ public void TestToBytes()
...
-				Behavior = 4,
-				IsLinkedGame = true,
-				IsHeroQuest = false,
-				WasGivenFreeRing = true
-			};
-
-			Assert.AreNotEqual(DesiredSecret, s2);
-			Assert.AreNotEqual(DesiredSecret, null);
-			Assert.AreNotEqual(DesiredSecret, "");
+			Assert.That(DesiredSecret.Equals(new GameSecret()), Is.False);
```
Comment (kabili207) [inline test/GameSecretTest.cs]: Change these to use `Is.Not.EqualTo` instead.
For example:
```csharp
Assert.That(DesiredSecret, Is.Not.EqualTo(new GameSecret());
```

**Pull request merged**